pkg/wasm/imagefetcher.go

```diff
@@ -311,3 +319,39 @@ func extractOCIArtifactImage(img v1.Image) ([]byte, error) {
 	}
 	return ret, nil
 }
+
+type wasmKeyChain struct {
+	data []byte
+}
+
+// Resolve an image reference to a credential.
+// The function code is borrowed from https://github.com/google/go-containerregistry/blob/v0.8.0/pkg/authn/keychain.go#L65,
+// by making it take dockerconfigjson directly as bytes instead of reading from files.
+func (k *wasmKeyChain) Resolve(target authn.Resource) (authn.Authenticator, error) {
+	reader := bytes.NewReader(k.data)
```

Comment:
This is not safe without https://github.com/docker/cli/pull/3434. Maybe we can add an explicit check for `null` in the mean time?

Reply:
Thanks for the pointer! Added an explicit check. Let me know if I did it right.